src/main/java/frc/lib/io/vision/VisionIOPhotonVision.java

```diff
@@ -15,123 +15,89 @@
-                var multitagResult = result.multitagResult.get();
-
-                // Calculate robot pose
-                Transform3d fieldToCamera = multitagResult.estimatedPose.best;
-                Transform3d fieldToRobot = fieldToCamera.plus(robotToCamera.inverse());
-                Pose3d robotPose =
-                    new Pose3d(fieldToRobot.getTranslation(), fieldToRobot.getRotation());
-
-                // Calculate average tag distance
-                double totalTagDistance = 0.0;
-                for (var target : result.targets) {
-                    totalTagDistance += target.bestCameraToTarget.getTranslation().getNorm();
-                }
+        List<PoseObservation> estimates = new ArrayList<>();
+        Set<Integer> tagIDs = new HashSet<>();
 
-                // Add tag IDs
-                tagIds.addAll(multitagResult.fiducialIDsUsed);
-
-                // Add observation
-                poseObservations.add(
-                    new PoseObservation(
-                        result.getTimestampSeconds(), // Timestamp
-                        robotPose, // 3D pose estimate
-                        multitagResult.estimatedPose.ambiguity, // Ambiguity
-                        multitagResult.fiducialIDsUsed.size(), // Tag count
-                        totalTagDistance / result.targets.size(), // Average tag distance
-                        PoseObservationType.PHOTONVISION)); // Observation type
+        poseEstimator.addHeadingData(timestampedHeading.timestamp().in(Seconds),
+            timestampedHeading.get());
+        for (PhotonPipelineResult result : camera.getAllUnreadResults()) {
+            if (!result.hasTargets()) {
+                continue;
+            }
 
-            } else if (!result.targets.isEmpty()) { // Single tag result
-                var target = result.targets.get(0);
+            Optional<EstimatedRobotPose> optionalEstimate = poseEstimator.update(result);
+            if (optionalEstimate.isEmpty()) {
+                continue;
+            }
 
-                // Calculate robot pose
-                var tagPose = aprilTagLayout.getTagPose(target.fiducialId);
-                if (tagPose.isPresent()) {
-                    Transform3d fieldToTarget = new Transform3d(tagPose.get().getTranslation(),
-                        tagPose.get().getRotation());
-                    Transform3d cameraToTarget = target.bestCameraToTarget;
-                    Transform3d fieldToCamera = fieldToTarget.plus(cameraToTarget.inverse());
-                    Transform3d fieldToRobot = fieldToCamera.plus(robotToCamera.inverse());
-                    Pose3d robotPose =
-                        new Pose3d(fieldToRobot.getTranslation(), fieldToRobot.getRotation());
+            EstimatedRobotPose estimate = optionalEstimate.get();
 
-                    // Add tag ID
-                    tagIds.add((short) target.fiducialId);
+            int tagCount = estimate.targetsUsed.size();
```

Comment:
Is there a reason not to declare this variables in advance? Or is it to ensure they don't have stale values?